templates/terraform/post_create/set_computed_name.erb

```diff
@@ -0,0 +1,8 @@
+// `name` is autogenerated from the api so needs to be set post-create
+name, ok := res["name"]
+if !ok {
+  return fmt.Errorf("Create response didn't contain critical fields. Create may not have succeeded.")
+}
+
+d.Set("name", name.(string))
+d.SetId(name.(string))
```

Comment:
i thought we were going to set the id to the display name so that we didn't have to have custom code around it

Reply:
You are correct and I forgot to update you after I ran into a couple different problems: I'm stuck using custom code in order to set `name` as a post-create anyway. When importing the resource we need to have a way to construct the GET url and right now the import logic is using the id. So I think for now the id has to contain enough information to construct a the resource url without heavy modification to the import logic.

Also I was rushing on Friday and didn't realize I had broke my import tests because the import logic also has an assumption that `/`'s have significance in import_id and will parse them out. So I'm going to just go back to commenting as much of this as possible into hawk's old strategy :(